### PR TITLE
Button restProps and non-mutating mods

### PR DESCRIPTION
**Pull request by asci**
## What
- Button component `restProps` for data attributes for simple button fix
- Don't mutate props.mods (fix for #77 ): slice mods if provided or just use a new array

## Where
- Diff

## Tests
- 100% covereage
- added new snapshots
- updated old ones

### CODE REVIEW

File: components/autoComplete/autoCompleteItem.js [hunk 1]
```diff
@@ -28,12 +28,12 @@ class AutoCompleteItem extends Component {
     const {
       children,
       dataAttrs = {},
-      mods = [],
       onClick,
       isActive,
       isTitle,
       ...otherProps
     } = this.props;
+    const mods = this.props.mods ? this.props.mods.slice() : [];
```
Comment (reaktivo) [inline components/autoComplete/autoCompleteItem.js]: I commented something similar in another PR, but seems to me that the following could be a good convention

    getClassNamesWithMods('ui-button', [
      isTitle && 'title',
      isActive && 'active',
    ]);

**Comment by asci** [comment]
Any comments?

File: components/button/button.js [hunk 2]
```diff
@@ -7,9 +7,19 @@ const { PropTypes } = React;
...
+    href,
+    onClick,
+    type,
+    variation,
+    disabled,
+    dataAttrs = {},
+  } = props;
   const restProps = getDataAttributes(dataAttrs);
-
+  const mods = props.mods ? props.mods.slice() : [];
```
Comment (reaktivo) [inline components/button/button.js]: I think it's a bad idea to have an arbitrary mods prop being passed here, specially considering there's no documentation for it and from what I see, should only affect internal behaviour (via the size and variation props)

Comment (asci) [inline components/button/button.js]: here is the docs for `mods`:
https://github.com/Travix-International/travix-ui-kit/pull/78/files/e6642dab2b515c3a613b7f0620ec86d6e3e41e4d#diff-22fbe4ecce9fc18f113d7f3df5fa1cc7R105 I guess the concept of mods is unclear: the mods property is for custom class selectors which developers need sometimes. And also we use it internally in components.

You can check how it works here https://travix-international.github.io/travix-ui-kit/#!/Button/1 there is an interactive playground.

The idea and the scope of this PR are to fix mutation of `mods` in all components, and not to change the way how they are work.

Comment (reaktivo) [inline components/button/button.js]: That was my point ` mods: PropTypes.arrayOf(PropTypes.string),` doesn't express at all how a developer would use it. It doesn't matter though, if we're moving to css-modules soon. And you're right about this is out of scope for this PR.
